### ISSUE TITLE
Status of the VJ Makers’ website

### ISSUE THREAD

**Issue by demoneaux** [vjmakers/vjmakers.github.io#2]
## VJ Makers’ Sites

| | Link |
|---|---|
| Website | https://vjmakers.github.io/ *([Source Code](https://github.com/vjmakers/vjmakers.github.io))* |
| Instagram | https://www.instagram.com/vjmakers/ |
| YouTube | https://www.youtube.com/channel/UCdF_Fxfn2WNCpO5eB_UjD_w |
| GitHub | https://github.com/vjmakers |

## Current status of VJ Maker’s site

Right now it's a pretty simple website made up of a few base files for the layout and design, and files for each page. It’s currently built by hand, using HTML, CSS, Jekyll and a few other technologies.

The first thing we should decide is whether to continue to use this setup, or switch to a separate system like WordPress or Blogger.

## Components of the website

| Component | Description |
|---|---|
| [HTML](http://marksheet.io/html-basics.html) | Simple markup language used to display content of the site |
| [CSS](http://marksheet.io/css-basics.html) | Used to style/design the site |
| [Jekyll](https://jekyllrb.com/) | Simple static site generator - converts our templates into a proper site |
| [Markdown](https://daringfireball.net/projects/markdown/)| Allows people to write text which can be formatted into HTML easily |
| [GitHub](https://github.com/) | Online storage for code |
| [GitHub Pages](https://pages.github.com/) | Simple way to host pages on GitHub |

**Comment by demoneaux** [comment]
/cc @casualvisitor @wei2912 @knowyourenemy

What do y’all think? Just look through this and after that we can discuss with Mr Ho and Joshua.

**Comment by wei2912** [comment]
@demoneaux what exactly is this for? Status wise, I think we should stick
to Jekyll, but with Ruby plugins (meaning we build on a separate server,
similar to what I do on my website). It's Makers Club so we should probably
make good use of the flexibility that Jekyll offers us as opposed to using
WordPress or Blogger with way less flexibility.

**Comment by demoneaux** [comment]
> @demoneaux what exactly is this for?

Just listing our options, for us to consider.

> Status wise, I think we should stick to Jekyll

Yep.

> but with Ruby plugins (meaning we build on a separate server,
> similar to what I do on my website)

Currently it's built on Travis servers and pushed to GitHub pages.

**Comment by wei2912** [comment]
@demoneaux There's no separate repository for the generated files? Or has
GitHub Pages changed

**Comment by demoneaux** [comment]
@wei2912 nope, but instead of using a separate repo I use one branch for the source and the `master` branch for the generated code.

**Comment by wei2912** [comment]
@demoneaux I see, that's a pretty nice solution. I'll push my edits soon.
